Configuration/Base.xcconfig

```diff
@@ -66,17 +66,12 @@ MACOSX_DEPLOYMENT_TARGET_1400 = 10.13;
 MACOSX_DEPLOYMENT_TARGET_1500 = 10.14;
 MACOSX_DEPLOYMENT_TARGET = $(MACOSX_DEPLOYMENT_TARGET_$(XCODE_VERSION_MAJOR));
 WATCHOS_DEPLOYMENT_TARGET_1400 = 4.0;
-WATCHOS_DEPLOYMENT_TARGET_1500 = 5.0;
+WATCHOS_DEPLOYMENT_TARGET_1500 = 4.0;
```

Comment:
I double-checked exactly what the minimum versions are with Xcode 15 and for whatever reason it turned out that they bumped the minimum iOS and tvOS versions but not the minimum watchOS version.

I left all the logic for setting deployment targets based on Xcode major version in place even though it's currently not relevant because Xcode 16 is presumably going to bump the minimum supported versions and we'd just have to restore it in a month.